File: src/dataset/javascript/triangle_exercism-ThomasZumsteg/claude/ClaudeSonnet4_triangle_v2.js
/* Classifies triangles */
var Triangle = function(a,b,c) { 
    // Validate inputs early to avoid unnecessary sorting
    if (a <= 0 || b <= 0 || c <= 0) 
        throw "Only positive length sides are allowed";
    
    // Sort once during construction
    this.triangle = [a,b,c].sort((x,y) => x - y);
    
    // Cache validation results
    var min = this.triangle[0];
    var mid = this.triangle[1];
    var max = this.triangle[2];
    
    if (min + mid <= max)
        throw "Not a valid triangle";
}

Triangle.prototype.kind = function() {
    var a = this.triangle[0];
    var b = this.triangle[1];
    var c = this.triangle[2];

    // Use strict equality for faster comparison
    if (a === b) {
        return b === c ? "equilateral" : "isosceles";
    }
    
    return b === c ? "isosceles" : "scalene";
};

module.exports = Triangle;

File: src/dataset/Java/triangle_exercism-ThomasZumsteg/claude/ClaudeSonnet4_Triangle_v2.java
public class Triangle {
    private double a, b, c;

    public Triangle(double a, double b, double c) throws TriangleException {
        // Sort manually to avoid array allocation and Arrays.sort overhead
        if (a > b) {
            if (b > c) {
                // a > b > c
                this.c = a; this.b = b; this.a = c;
            } else if (a > c) {
                // a > c >= b
                this.c = a; this.b = c; this.a = b;
            } else {
                // c >= a > b
                this.c = c; this.b = a; this.a = b;
            }
        } else {
            if (a > c) {
                // b >= a > c
                this.c = b; this.b = a; this.a = c;
            } else if (b > c) {
                // b > c >= a
                this.c = b; this.b = c; this.a = a;
            } else {
                // c >= b >= a
                this.c = c; this.b = b; this.a = a;
            }
        }
        
        if (this.a < 0 || this.a + this.b <= this.c) {
            throw new TriangleException();
        }
    }

    public TriangleKind getKind() {
        if (a == b) {
            return b == c ? TriangleKind.EQUILATERAL : TriangleKind.ISOSCELES;
        } else {
            return b == c ? TriangleKind.ISOSCELES : TriangleKind.SCALENE;
        }
    }
}